### PR TITLE
Enhance e2e test about orders

### PR DESCRIPTION
**Pull request by HoOngEe**
Added some edge cases for transactions with orders. And now
the order transactions with fee recipients are included in the test suits.

### CODE REVIEW

File: test/src/e2e.long/orders.test.ts [hunk 1]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                        spentQuantity: 100,
+                        inputIndices: [0, 2],
+                        outputIndices: [0, 1, 2, 5]
+                    });
+                await node.signTransactionInput(transferTx2, 0);
+                await node.signTransactionInput(transferTx2, 1);
+                await node.signTransactionInput(transferTx2, 2);
+
+                const results2 = await node.sendAssetTransaction(transferTx2);
+                expect(results2!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(results2).deep.equal([false]);
```

Comment (sgkim126) [reply test/src/e2e.long/orders.test.ts]: By the way, why `result2`?

File: test/src/e2e.long/orders.test.ts [hunk 2]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                        inputIndices: [2, 3],
+                        outputIndices: [5, 6, 7, 8]
+                    });
+                await node.signTransactionInput(transferTx, 0);
+                await node.signTransactionInput(transferTx, 1);
+                await node.signTransactionInput(transferTx, 2);
+                await node.signTransactionInput(transferTx, 3);
+
+                const results = await node.sendAssetTransaction(transferTx);
+                expect(results!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(results).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 3]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                            recipient: bobAddress,
+                            quantity: 5000,
+                            assetType: bronze.assetType,
+                            shardId: 0
+                        }
+                    );
+                await node.signTransactionInput(splitTx, 0);
+
+                const splitResults = await node.sendAssetTransaction(splitTx);
+                expect(splitResults!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(splitResults).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 4]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                        inputIndices: [2, 3],
+                        outputIndices: [4, 5, 6, 7]
+                    });
+                await node.signTransactionInput(transferTx, 0);
+                await node.signTransactionInput(transferTx, 1);
+                await node.signTransactionInput(transferTx, 2);
+                await node.signTransactionInput(transferTx, 3);
+
+                const results = await node.sendAssetTransaction(transferTx);
+                expect(results!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(splitResults).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 5]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                            recipient: bobAddress,
+                            quantity: 5000,
+                            assetType: bronze.assetType,
+                            shardId: 0
+                        }
+                    );
+                await node.signTransactionInput(splitTx, 0);
+
+                const splitResults = await node.sendAssetTransaction(splitTx);
+                expect(splitResults!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(splitResults).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 6]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                        order,
+                        spentQuantity: 100,
+                        inputIndices: [0, 2],
+                        outputIndices: [0, 1, 2, 5]
+                    });
+                await node.signTransactionInput(transferTx2, 0);
+                await node.signTransactionInput(transferTx2, 1);
+                await node.signTransactionInput(transferTx2, 2);
+
+                const results2 = await node.sendAssetTransaction(transferTx2);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(results2).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 7]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(results2).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts [hunk 8]
```diff
@@ -1732,6 +2153,601 @@ describe("orders", function() {
...
+                            recipient: aliceAddress,
+                            quantity: 1000,
+                            assetType: bronze.assetType,
+                            shardId: 0
+                        }))
+                    );
+                await node.signTransactionInput(splitTx, 0);
+
+                const splitResults = await node.sendAssetTransaction(splitTx);
+                expect(splitResults!.length).to.equal(1);
```
Comment (sgkim126) [inline test/src/e2e.long/orders.test.ts]:
```ts
expect(splitResults).deep.equal([true]);
```

File: test/src/e2e.long/orders.test.ts (hunk 1 above)
Comment (HoOngEe) [reply test/src/e2e.long/orders.test.ts]: All meaningless 2s are removed.
